File: dotnet/src/Botas/TeamsChannelData.cs
using System.Text.Json.Serialization;

namespace Botas;

public class TenantInfo
{
    [JsonPropertyName("id")] public string? Id { get; set; }
    [JsonExtensionData] public ExtendedPropertiesDictionary Properties { get; set; } = [];
}

public class ChannelInfo
{
    [JsonPropertyName("id")] public string? Id { get; set; }
    [JsonPropertyName("name")] public string? Name { get; set; }
    [JsonExtensionData] public ExtendedPropertiesDictionary Properties { get; set; } = [];
}

public class TeamInfo
{
    [JsonPropertyName("id")] public string? Id { get; set; }
    [JsonPropertyName("name")] public string? Name { get; set; }
    [JsonPropertyName("aadGroupId")] public string? AadGroupId { get; set; }
    [JsonExtensionData] public ExtendedPropertiesDictionary Properties { get; set; } = [];
}

public class MeetingInfo
{
    [JsonPropertyName("id")] public string? Id { get; set; }
    [JsonExtensionData] public ExtendedPropertiesDictionary Properties { get; set; } = [];
}

public class NotificationInfo
{
    [JsonPropertyName("alert")] public bool? Alert { get; set; }
    [JsonExtensionData] public ExtendedPropertiesDictionary Properties { get; set; } = [];
}

public class TeamsChannelData
{
    [JsonPropertyName("tenant")] public TenantInfo? Tenant { get; set; }
    [JsonPropertyName("channel")] public ChannelInfo? Channel { get; set; }
    [JsonPropertyName("team")] public TeamInfo? Team { get; set; }
    [JsonPropertyName("meeting")] public MeetingInfo? Meeting { get; set; }
    [JsonPropertyName("notification")] public NotificationInfo? Notification { get; set; }
    [JsonExtensionData] public ExtendedPropertiesDictionary Properties { get; set; } = [];
}

File: python/packages/botas/src/botas/teams_channel_data.py
from __future__ import annotations

from pydantic import BaseModel, ConfigDict
from pydantic.alias_generators import to_camel


class _CamelModel(BaseModel):
    model_config = ConfigDict(
        alias_generator=to_camel,
        populate_by_name=True,
        extra="allow",
    )


class TenantInfo(_CamelModel):
    id: str | None = None


class ChannelInfo(_CamelModel):
    id: str | None = None
    name: str | None = None


class TeamInfo(_CamelModel):
    id: str | None = None
    name: str | None = None
    aad_group_id: str | None = None


class MeetingInfo(_CamelModel):
    id: str | None = None


class NotificationInfo(_CamelModel):
    alert: bool | None = None


class TeamsChannelData(_CamelModel):
    tenant: TenantInfo | None = None
    channel: ChannelInfo | None = None
    team: TeamInfo | None = None
    meeting: MeetingInfo | None = None
    notification: NotificationInfo | None = None
